Content.Server/Clothing/EntitySystems/ToggleableClothingSystem.cs

```diff
@@ -0,0 +1,174 @@
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<ToggleableClothingComponent, ToggleClothingEvent>(OnToggleClothing);
+            SubscribeLocalEvent<ToggleableClothingComponent, GetItemActionsEvent>(OnGetActions);
+        }
+
+        private void OnGetActions(EntityUid uid, ToggleableClothingComponent component, GetItemActionsEvent args)
+        {
+            if (component.ToggleAction != null && component.SelfToggle)
+            {
+                // logic for self toggled clothing like masks
+                args.Actions.Add(component.ToggleAction);
+                ResetSelfToggleClothing(uid, component);
+            }
+            else
+            {
+                // logic for toggleable clothing tied to another like hardsuits
+                if (component.ClothingUid == null || (args.SlotFlags & component.RequiredFlags) != component.RequiredFlags)
+                    return;
+
+                if (component.ToggleAction != null)
+                    args.Actions.Add(component.ToggleAction);
+            }
+        }
+
+        private void OnToggleClothing(EntityUid uid, ToggleableClothingComponent component, ToggleClothingEvent args)
+        {
+            // self toggled clothing, like masks
+            if (component.SelfToggle)
+            {
+                OnSelfToggleClothing(uid, component, args);
+                args.Handled = true;
+                return;
+            }
+
+            // continue on for standard toggleable clothing, like hardsuits
+            if (args.Handled || component.Container == null || component.ClothingUid == null)
+                return;
+
+            var parent = Transform(uid).ParentUid;
+            if (component.Container.ContainedEntity == null)
+                _inventorySystem.TryUnequip(parent, component.Slot);
+            else if (_inventorySystem.TryGetSlotEntity(parent, component.Slot, out var existing))
+            {
+                _popupSystem.PopupEntity(Loc.GetString("toggleable-clothing-remove-first", ("entity", existing)),
+                    args.Performer, Filter.Entities(args.Performer));
+            }
+            else
+                _inventorySystem.TryEquip(parent, component.ClothingUid.Value, component.Slot);
+
+            args.Handled = true;
+        }
+
+        private void OnSelfToggleClothing(EntityUid uid, ToggleableClothingComponent component, ToggleClothingEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            switch (component.Slot)
+            {
+                case "mask":
+                    if (TryEquip(uid, component, args, component.Slot))
+                    {
+                        component.IsToggled ^= true;
+                        ToggleMask(uid, component, args.Performer);
+                        //TODO: update visuals?
+                    }
+                    break;
+            }
+        }
```

Comment:
this should be handled in the action of the toggleableclothingcomp, not hardcoded